Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean

```diff
@@ -330,10 +330,151 @@ theorem ιMulti_succ_curryLeft {n : ℕ} (m : M) :
+  simp only [List.map_ofFn, Function.comp, map_apply_ι]
+
+@[simp]
+theorem map_comp_ιMulti {n : ℕ} (f : M →ₗ[R] N) :
+    (map f).toLinearMap.compAlternatingMap (ιMulti R n (M := M)) =
+    (ιMulti R n (M := N)).compLinearMap f := by
+  ext m
+  exact map_apply_ιMulti _ _
+
+@[simp]
+theorem map_id :
+    map LinearMap.id = AlgHom.id R (ExteriorAlgebra R M) :=
+  CliffordAlgebra.map_id 0
+
+@[simp]
+theorem map_comp_map (f : M →ₗ[R] N) (g : N →ₗ[R] N') :
+    AlgHom.comp (map g) (map f) = map (LinearMap.comp g f) :=
+  CliffordAlgebra.map_comp_map _ _
+
+@[simp]
+theorem ι_range_map_map (f : M →ₗ[R] N) :
+    Submodule.map (AlgHom.toLinearMap (map f)) (LinearMap.range (ι R (M := M))) =
+    Submodule.map (ι R) (LinearMap.range f) :=
+  CliffordAlgebra.ι_range_map_map _
+
+theorem toTrivSqZeroExt_comp_map [Module Rᵐᵒᵖ M] [IsCentralScalar R M] [Module Rᵐᵒᵖ N]
+    [IsCentralScalar R N] (f : M →ₗ[R] N) :
+    toTrivSqZeroExt.comp (map f) = (TrivSqZeroExt.map f).comp toTrivSqZeroExt := by
+  apply hom_ext
+  apply LinearMap.ext
+  intro x
+  unfold TrivSqZeroExt.map map
+  simp only [AlgHom.comp_toLinearMap, LinearMap.coe_comp, Function.comp_apply,
+    AlgHom.toLinearMap_apply, CliffordAlgebra.map_apply_ι, toTrivSqZeroExt_ι,
+    TrivSqZeroExt.lift_apply_inr, TrivSqZeroExt.inrHom_apply]
+  rfl
+
+theorem ιInv_comp_map (f : M →ₗ[R] N) :
+    ιInv.comp (map f).toLinearMap = f.comp ιInv := by
+  letI : Module Rᵐᵒᵖ M := Module.compHom _ ((RingHom.id R).fromOpposite mul_comm)
+  haveI : IsCentralScalar R M := ⟨fun r m => rfl⟩
+  letI : Module Rᵐᵒᵖ N := Module.compHom _ ((RingHom.id R).fromOpposite mul_comm)
+  haveI : IsCentralScalar R N := ⟨fun r m => rfl⟩
+  unfold ιInv
+  conv_lhs => rw [LinearMap.comp_assoc, ← AlgHom.comp_toLinearMap, toTrivSqZeroExt_comp_map,
+                AlgHom.comp_toLinearMap, ← LinearMap.comp_assoc, TrivSqZeroExt.snd_comp_map]
+
+open Function in
+/-- For a linear map `f` from `M` to `N`,
+`ExteriorAlgebra.map g` is a retraction of `ExteriorAlgebra.map f` iff
+`g` is a retraction of `f`. -/
+@[simp]
+lemma leftInverse_map_iff {f : M →ₗ[R] N} {g : N →ₗ[R] M} :
+    LeftInverse (map g) (map f) ↔ LeftInverse g f := by
+  refine ⟨fun h x => ?_, fun h x => ?_⟩
+  · simpa using h (ι _ x)
+  · exact CliffordAlgebra.leftInverse_map_of_leftInverse (Q₁ := 0) (Q₂ := 0)
+      { f with map_app' := fun _ => rfl } { g with map_app' := fun _ => rfl} h x
+
+/-- A morphism of modules that admits a linear retraction induces an injective morphism of
+exterior algebras. -/
+lemma map_injective {f : M →ₗ[R] N} (hf : ∃ (g : N →ₗ[R] M), g.comp f = LinearMap.id) :
+    Function.Injective (map f) :=
+  let ⟨_, hgf⟩ := hf; (leftInverse_map_iff.mpr (DFunLike.congr_fun hgf)).injective
+
+/-- A morphism of modules is surjective if and only the morphism of exterior algebras that it
+induces is surjective. -/
+lemma map_surjective_iff {f : M →ₗ[R] N} :
+    Function.Surjective f ↔ Function.Surjective (map f) := by
```

Comment:
This statement is backwards with respect to its name.
```suggestion
    Function.Surjective (map f) ↔ Function.Surjective f := by
```

Reply:
I'm not sure why, but ok.

Reply:
`foo_iff` is how statements of the form `Foo ↔ Bar` are named. Statements of the form `Bar ↔ Foo ` are called `bar_iff` or `bar_iff_foo`.

In this case, putting the `map` on the left also makes this a nice `simp` lemma.

Reply:
Ah, I see, thanks. Then I should actually make it a 'simp' lemma then.